airbyte-integrations/bases/base-typing-deduping/src/test/java/io/airbyte/integrations/base/destination/typing_deduping/AirbyteTypeTest.java

```diff
@@ -417,64 +421,55 @@ public void testInvalid() {
     invalidSchema.add("{}");
 
     for (final String schema : invalidSchema) {
-      assertEquals(AirbyteProtocolType.UNKNOWN, AirbyteType.fromJsonSchema(Jsons.deserialize(schema)));
+      assertEquals(UNKNOWN, fromJsonSchema(Jsons.deserialize(schema)));
     }
   }
 
   @Test
-  public void testChooseOneOf() {
-    // test ordering
+  public void testChooseUnion() {
+    final Map<Union, AirbyteType> unionToType = new HashMap<>();
 
-    OneOf o = new OneOf(ImmutableList.of(AirbyteProtocolType.STRING, AirbyteProtocolType.DATE));
-    assertEquals(AirbyteProtocolType.DATE, AirbyteTypeUtils.chooseOneOfType(o));
-
-    final Array a = new Array(AirbyteProtocolType.TIME_WITH_TIMEZONE);
-    o = new OneOf(ImmutableList.of(AirbyteProtocolType.TIMESTAMP_WITH_TIMEZONE, a));
-    assertEquals(a, AirbyteTypeUtils.chooseOneOfType(o));
+    final Array a = new Array(BOOLEAN);
 
     final LinkedHashMap<String, AirbyteType> properties = new LinkedHashMap<>();
-    properties.put("key1", AirbyteProtocolType.UNKNOWN);
-    properties.put("key2", AirbyteProtocolType.TIME_WITHOUT_TIMEZONE);
+    properties.put("key1", UNKNOWN);
+    properties.put("key2", INTEGER);
     final Struct s = new Struct(properties);
-    o = new OneOf(ImmutableList.of(AirbyteProtocolType.TIMESTAMP_WITHOUT_TIMEZONE, s));
-    assertEquals(s, AirbyteTypeUtils.chooseOneOfType(o));
-
-    // test exclusion
-
-    o = new OneOf(ImmutableList.of(AirbyteProtocolType.BOOLEAN, AirbyteProtocolType.INTEGER));
-    assertEquals(AirbyteProtocolType.INTEGER, AirbyteTypeUtils.chooseOneOfType(o));
 
-    o = new OneOf(ImmutableList.of(AirbyteProtocolType.INTEGER, AirbyteProtocolType.NUMBER, AirbyteProtocolType.DATE));
-    assertEquals(AirbyteProtocolType.NUMBER, AirbyteTypeUtils.chooseOneOfType(o));
+    unionToType.put(new Union(ImmutableList.of(s, a)), a);
+    unionToType.put(new Union(ImmutableList.of(NUMBER, a)), a);
+    unionToType.put(new Union(ImmutableList.of(INTEGER, s)), s);
+    unionToType.put(new Union(ImmutableList.of(NUMBER, DATE, BOOLEAN)), DATE);
+    unionToType.put(new Union(ImmutableList.of(INTEGER, BOOLEAN, NUMBER)), NUMBER);
+    unionToType.put(new Union(ImmutableList.of(BOOLEAN, INTEGER)), INTEGER);
 
-    o = new OneOf(ImmutableList.of(AirbyteProtocolType.BOOLEAN, AirbyteProtocolType.NUMBER, AirbyteProtocolType.STRING));
-    assertEquals(AirbyteProtocolType.STRING, AirbyteTypeUtils.chooseOneOfType(o));
+    unionToType.forEach((u, t) -> assertEquals(t, chooseUnionType(u)));
```

Comment:
nit: might want to do something with `assertAll` here

Reply:
I wrapped the line in an assertAll but lmk if that's not what you had in mind.